src/oneapitest/SYCLCore/ScopedSetDevice.h

```diff
@@ -0,0 +1,29 @@
+#ifndef HeterogeneousCore_SYCLUtilities_ScopedSetDevice_h
+#define HeterogeneousCore_SYCLUtilities_ScopedSetDevice_h
+
+#include <CL/sycl.hpp>
+#include <dpct/dpct.hpp>
+
+namespace cms {
+  namespace sycl {
+    class ScopedSetDevice {
+    public:
+      explicit ScopedSetDevice(int newDevice) {
+        prevDevice_ = dpct::dev_mgr::instance().current_device_id();
+        dpct::dev_mgr::instance().select_device(newDevice);
```

Comment:
I see SYCL also has the notion of the "current device" (is it per thread?). Does something actually need it? Or can the device be always given explicitly (or implicitly via the queue) in the API calls?

Reply:
Actually this comes from `dpct`, the Intel compatibility library for porting CUDA code (which I'll slowly try to migrate away from).

Given that all operations require a queue, and a queue knows which devices it runs on, I'd simply remove the "current device" from everywhere.

Reply:
> Given that all operations require a queue, and a queue knows which devices it runs on, I'd simply remove the "current device" from everywhere.

I would prefer that as well.